lib/modules/commentNavigator.js

```diff
@@ -241,13 +253,17 @@ function changeCategory() {
 	if (index === -1) {
 		return;
 	}
-	currentCategory = (commentNav().select.options.item(index).value: any);
-	if (currentCategory !== '') {
+	const newCategory = (commentNav().select.options.item(index).value: any);
+	if (newCategory !== currentCategory) {
+		resetNavigator(newCategory);
+	}
+	if (newCategory !== '') {
 		getPostsByCategory();
```

Comment:
Is must also be called when there's no category selected, in order to avoid too much logic.

Reply:
I think I found it like this, but ok!